onnxruntime/core/platform/posix/env.cc

```diff
@@ -143,47 +143,62 @@ class PosixThread : public EnvThread {
   PosixThread(const ORTCHAR_T* name_prefix, int index,
               unsigned (*start_address)(int id, Eigen::ThreadPoolInterface* param), Eigen::ThreadPoolInterface* param,
               const ThreadOptions& thread_options) {
-    pthread_attr_t attr;
-    int s = pthread_attr_init(&attr);
-    if (s != 0) {
-      auto[err_no, err_msg] = GetSystemError();
-      ORT_THROW("pthread_attr_init failed, error code: ", err_no, " error msg: ", err_msg);
-    }
-    if (thread_options.stack_size > 0) {
-      s = pthread_attr_setstacksize(&attr, thread_options.stack_size);
+    custom_create_thread_fn = thread_options.custom_create_thread_fn;
+    custom_thread_creation_options = thread_options.custom_thread_creation_options;
+    custom_join_thread_fn = thread_options.custom_join_thread_fn;
+
+    if (custom_create_thread_fn) {
+      hThread = (pthread_t)custom_create_thread_fn(custom_thread_creation_options, CustomThreadMain, new Param{name_prefix, index, start_address, param, thread_options});
```

Comment:
The handle returned by custom_create_thread_fn might not be a pthread_t instance and the cast may not be valid. Any reason we have to cast this to pthread_t? Same comment for the windows part.

Reply:
Sure, defined a custom thread handle of type void* to represent a foreign thread.